From 9d0d5e63d09fb46b8cbeb24b154c3c2360dbe7c6 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Tue, 30 Sep 2014 14:43:39 +0100
Subject: [PATCH 04/14] Remove redundancy in UnixCommand's exe checking
 functions

Check the result of UnixCommand::discoverBinaryPath() not
being empty in UnixCommand::hasTheExecutable() instead of
repeating much the same code.
---
 src/unixcommand.cpp | 18 +++++++-----------
 1 file changed, 7 insertions(+), 11 deletions(-)

diff --git a/src/unixcommand.cpp b/src/unixcommand.cpp
index 9203eae..4dcfe42 100644
--- a/src/unixcommand.cpp
+++ b/src/unixcommand.cpp
@@ -81,6 +81,7 @@ QString UnixCommand::runCurlCommand(const QString& commandToRun){
 QString UnixCommand::discoverBinaryPath(const QString& binary){
   QProcess proc;
   proc.setProcessEnvironment(getEnv());
+  proc.setProcessChannelMode(QProcess::MergedChannels);
 
   proc.start("/bin/sh -c \"which " + binary + "\"");
   proc.waitForFinished();
@@ -89,6 +90,11 @@ QString UnixCommand::discoverBinaryPath(const QString& binary){
   proc.close();
   res = res.remove('\n');
 
+  if (res.count("which"))
+      res.clear();
+
+  if (!res.isEmpty()) return res;
+
   //If it still didn't find it, try "/sbin" dir...
   if (res.isEmpty()){
     QFile fbin("/sbin/" + binary);
@@ -507,17 +513,7 @@ bool UnixCommand::doInternetPingTest()
  */
 bool UnixCommand::hasTheExecutable( const QString& exeName )
 {
-  QProcess proc;
-  proc.setProcessChannelMode(QProcess::MergedChannels);
-  QString sParam = "\"which " + exeName + "\"";
-  proc.start("/bin/sh -c " + sParam);
-  proc.waitForFinished();
-
-  QString out = proc.readAllStandardOutput();
-  proc.close();
-
-  if (out.isEmpty() || out.count("which") > 0) return false;
-  else return true;
+  return (!discoverBinaryPath(exeName).isEmpty());
 }
 
 /*
-- 
2.2.1

